network/src/peers.rs

```diff
@@ -0,0 +1,192 @@
+use super::*;
+use async_trait::async_trait;
+use eyre::{eyre, Result};
+use serde_tc::http::*;
+use serde_tc::{serde_tc_full, StubCall};
+use simperby_core::serde_spb;
+use simperby_core::BlockHeader;
+use simperby_core::FinalizationInfo;
+use std::collections::BTreeMap;
+use std::sync::Arc;
+use tokio::fs::File;
+use tokio::io::AsyncReadExt;
+use tokio::io::AsyncWriteExt;
+use tokio::sync::RwLock;
+
+#[derive(Debug)]
+struct PeerStorage {
+    path: String,
+}
+
+impl PeerStorage {
+    pub async fn new(path: &str) -> Result<Self> {
+        Ok(Self {
+            path: path.to_owned(),
+        })
+    }
+
+    pub async fn write(&mut self, peers: Vec<Peer>) -> Result<()> {
+        let _ = tokio::fs::remove_file(&self.path).await;
+        let mut file = File::create(&self.path).await?;
+        file.write_all(serde_spb::to_string(&peers)?.as_bytes())
+            .await?;
+        file.flush().await?;
+        Ok(())
+    }
+
+    pub async fn read(&self) -> Result<Vec<Peer>> {
+        let mut file = File::open(&self.path).await?;
+        let peers: Vec<Peer> = serde_spb::from_str(&{
+            let mut buf = String::new();
+            file.read_to_string(&mut buf).await?;
+            buf
+        })?;
+        Ok(peers)
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct PingResponse {
+    pub last_finalized_block_header: BlockHeader,
+    pub public_key: PublicKey,
+    pub timestamp: Timestamp,
+    pub msg: String,
+}
+
+#[serde_tc_full]
+pub(super) trait PeerRpcInterface: Send + Sync + 'static {
+    /// Requests to response some packets.
+    async fn ping(&self) -> Result<PingResponse, String>;
+    /// Requests to response the port map of this node
+    async fn port_map(&self) -> Result<BTreeMap<String, u16>, String>;
+}
+
+pub struct PeerRpcImpl {
+    internal: Arc<RwLock<Peers>>,
```

Comment:
It's just my opinion, how about changing this name to just `peers`?

Reply:
fixed